Indicators/ValueAtRisk.cs

```diff
@@ -0,0 +1,133 @@
+        public override int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Gets a flag indicating when the indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => _dataPoints.Samples >= WarmUpPeriod;
+
+        /// <summary>
+        /// Creates a new ValueAtRisk indicator with a specified period and confidence level
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="period">Historical lookback period in days</param>
+        /// <param name="confidenceLevel">Confidence level for VaR calculation</param>
+        public ValueAtRisk(string name, int period, double confidenceLevel) 
+            : base(name, period)
+        {
+            if (period < 3)
+            {
+                throw new ArgumentException($"Period parameter for ValueAtRisk indicator must be greater than 2 but was {period}");
+            }
+
+            WarmUpPeriod = 3;
+            _confidenceLevel = confidenceLevel;
+
+            _dataPoints = new RollingWindow<decimal>(2);
+            _returns = new RollingWindow<double>(period);
+        }
+
+        /// <summary>
+        /// Creates a new ValueAtRisk indicator with a specified period and confidence level
+        /// </summary>
+        /// <param name="period">Historical lookback period in days</param>
+        /// <param name="confidenceLevel">Confidence level for VaR calculation</param>
+        public ValueAtRisk(int period, double confidenceLevel)
+            : this($"VaR({period}, {confidenceLevel})", period, confidenceLevel)
+        {
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+        /// <param name="window">The window of data held in this indicator</param>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint> window, IndicatorDataPoint input)
+        {
+            _dataPoints.Add(input.Value);
+
+            if (_dataPoints.Samples > 1)
+            {
+                _returns.Add(GetNewReturn(_dataPoints));
+            }
+            
+            if (_returns.Count < 2)
+            {
+                return 0m;
+            }
+
+            var mean = _returns.Mean();
+            var standardDeviation = _returns.StandardDeviation();
+            return (decimal)Normal.InvCDF(mean, standardDeviation, 1 - _confidenceLevel);
+        }
+
+        /// <summary>
+        /// Computes the returns by comparing the new data point and the previous data point
+        /// </summary>
+        /// <param name="rollingWindow">The collection of data points</param>
+        /// <returns>The return value</returns>
+        private static double GetNewReturn(RollingWindow<decimal> rollingWindow)
```

Comment:
I think cloud replace the manual return calculation by using the RateOfChange indicator?

Reply:
Thanks - done